recipes/tls-create-secure-pair-to-tls-socket/src/workflow.ts

```diff
@@ -0,0 +1,172 @@
+}
+
+function buildOptions(a?: string | null, b?: string | null, c?: string | null, d?: string | null): string {
+	const kv: string[] = [];
+	if (a) kv.push(`secureContext: ${a}`);
+	if (b) kv.push(`isServer: ${b}`);
+	if (c) kv.push(`requestCert: ${c}`);
+	if (d) kv.push(`rejectUnauthorized: ${d}`);
+	return `{ ${kv.join(', ')} }`;
+}
+
+function getText(node: SgNode<JS> | undefined): string | null {
+	const t = node?.text()?.trim();
+	return t || null;
+}
+
+function unique<T>(arr: T[]): T[] {
+	return Array.from(new Set(arr));
+}
+
+function renamePairAssignedVariables(rootNode: SgNode<JS>, bindings: string[]): Edit[] {
+	const edits: Edit[] = [];
+	for (const binding of bindings) {
+		const decls = rootNode.findAll({
+			rule: {
+				kind: 'variable_declarator',
+				has: {
+					field: 'value',
+					kind: 'call_expression',
+					pattern: `${binding}($$$ARGS)`,
+				},
+			},
+		});
+		for (const decl of decls) {
+			const name = decl.field('name');
+			if (name && name.kind() === 'identifier' && name.text() === 'pair') {
+				edits.push(name.replace('socket'));
+			}
+		}
+	}
+	return edits;
+}
+
+
+function rewriteTlsImports(rootNode: SgNode<JS>): Edit[] {
+	const edits: Edit[] = [];
+	const tlsStmts = rootNode.findAll({
+		rule: {
+			any: [
+				{ pattern: `import $$ANY from 'tls'` },
+				{ pattern: `import $$ANY from 'node:tls'` },
+				{ pattern: `const $$ANY = require('tls')` },
+				{ pattern: `const $$ANY = require('node:tls')` },
+			],
+		},
+	});
+
+	for (const stmt of tlsStmts) {
+		const code = stmt.text();
+		if (/\bimport\s+\*\s+as\s+\w+\s+from\s+['"](tls|node:tls)['"]/.test(code)) continue;
+		{
+			const m = code.match(
+				/import\s+(?:(?<def>[\w$]+)\s*,\s*)?\{\s*(?<names>[^}]*)\s*\}\s*from\s*['"](tls|node:tls)['"]\s*;?/,
+			);
+			if (m?.groups) {
+				const def = m.groups.def || '';
+				const namesRaw = m.groups.names || '';
+				const names = namesRaw
+					.split(',')
+					.map(s => s.trim())
+					.filter(Boolean)
+					.map(s => s.replace(/\s+as\s+\w+$/, ''));
+
+				if (names.includes('createSecurePair')) {
+					const kept = Array.from(
+						new Set(names.filter(n => n !== 'createSecurePair').concat('TLSSocket')),
+					);
+					const left = def ? `${def}, ` : '';
+					const rebuilt = `import ${left}{ ${kept.join(', ')} } from 'node:tls';`;
+					edits.push(stmt.replace(rebuilt));
+					continue;
+				}
+			}
+		}
+		{
+			const m = code.match(
+				/const\s*\{\s*(?<names>[^}]*)\s*\}\s*=\s*require\(\s*['"](tls|node:tls)['"]\s*\)\s*;?/,
+			);
+			if (m?.groups) {
+				const namesRaw = m.groups.names || '';
+				const names = namesRaw
+					.split(',')
+					.map(s => s.trim())
+					.filter(Boolean);
+
+				if (names.includes('createSecurePair')) {
+					const kept = Array.from(
+						new Set(names.filter(n => n !== 'createSecurePair').concat('TLSSocket')),
+					);
+					const src = /node:tls/.test(code) ? 'node:tls' : 'tls';
+					const rebuilt = `const { ${kept.join(', ')} } = require('${src}');`;
```

Comment:
```suggestion

					const rebuilt = `const { ${kept.join(', ')} } = require('node:tls');`;
```

we can enforce `node:tls` prefix

Reply:
👍